component/prometheus/fanout.go

```diff
@@ -37,10 +38,18 @@ func NewFanout(children []storage.Appendable, componentID string, register prome
 		Help: "Write latency for sending to direct and indirect components",
 	})
 	_ = register.Register(wl)
+
+	s := prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "agent_prometheus_fanout_samples_count",
```

Comment:
Naming nits:

* Counters should end in `_total`, see [Metric and label naming](https://prometheus.io/docs/practices/naming/#metric-and-label-naming) from Prometheus' docs for information on how to name metrics. 
* `fanout` feels more of an implementation detail and might not make sense to users. What about `agent_prometheus_forwarded_samples_total`?

Reply:
Thank you, `agent_prometheus_forwarded_samples_total` sounds good, I'll change it.

I got the "_count" suffix from a slide from the "Playlist: Intro to querying metrics with PromQL" learning path on Docebo:
![Screenshot 2023-01-17 at 22 10 11](https://user-images.githubusercontent.com/5834788/214007230-d35d6a50-c87b-4683-8cd9-129c13565837.png)

Should I ask the creators of the course to remove "_count" from it?

Reply:
Yes, that information seems to be just wrong. See the [OpenMetrics standard](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes) for rules on when to use which suffix.